### PR TITLE
conformance image e2e: stop using kind from HEAD and simplify script

### PR DESCRIPTION
**Pull request by BenTheElder**
also fix build for new go-runner

includes relevant portion of https://github.com/kubernetes/test-infra/pull/13159, squashes other bugs, fixes shellcheck issues.

### CODE REVIEW

**Comment by BenTheElder** [comment]
/assign @spiffxp @neolit123 
cc @johnSchnake

File: experiment/kind-conformance-image-e2e.sh [hunk 1]
```diff
@@ -87,19 +75,20 @@ build() {
...
 # up a cluster with kind
 create_cluster() {
     # create the config file
     cat <<EOF > "${ARTIFACTS}/kind-config.yaml"
 # config for 1 control plane node and 2 workers
 # necessary for conformance
-kind: Config
-apiVersion: kind.sigs.k8s.io/v1alpha2
+kind: Cluster
+apiVersion: kind.sigs.k8s.io/v1alpha3
```
Comment (BenTheElder) [inline experiment/kind-conformance-image-e2e.sh]: this was deprecated for a long time (since v0.2.0), updated.

Comment (BenTheElder) [reply experiment/kind-conformance-image-e2e.sh]: er: and removed in v0.4.0, so this fix is required...

**Comment by johnSchnake** [comment]
```
ERROR: security jobs changed, run hack/update-config.sh
```

💯 Thanks @BenTheElder !

**Comment by spiffxp** [comment]
/approve
I trust that someone else can re-lgtm once @BenTheElder addresses the security jobs

**Comment by BenTheElder** [comment]
I also had to rebase to pick up bazel fixes in order to run the update config script. Should be fixed now.

**Pull request merged**